server/rpc/gateway.go

```diff
@@ -107,13 +108,21 @@ func (s *gatewayServer) listenAndServe(mux *runtime.ServeMux) error {
 		WriteTimeout: time.Duration(cfg.API.WriteTimeout) * time.Second,
 		ReadTimeout:  time.Duration(cfg.API.ReadTimeout) * time.Second,
 	}
+	s.svr = svr
+
 	return svr.ListenAndServe()
 }
 
 func (s *gatewayServer) Close() error {
 	log.Info("Close API server")
+	if err := s.svr.Close(); err != nil {
```

Comment:
it would be better to use `Shutdown()` instead of `Close()` if you want graceful shutdown.

Reply:
I fixed it :)